esmvaltool/utils/cmorizers/obs/osi_common.py

```diff
@@ -0,0 +1,162 @@
+
+logger = logging.getLogger(__name__)
+
+
+def cmorize_osi(in_dir, out_dir, cfg, hemisphere):
+    """
+    Cmorize OSI-450 or OSI-407 dataset
+    """
+    cmor_table = cfg['cmor_table']
+    glob_attrs = cfg['attributes']
+
+    logger.info("Starting cmorization for Tier%s OBS files: %s",
+                glob_attrs['tier'], glob_attrs['dataset_id'])
+    logger.info("Input data from: %s", in_dir)
+    logger.info("Output will be written to: %s", out_dir)
+
+    # run the cmorization
+    first_run = True
+    for var, vals in cfg['variables'].items():
+        for year in os.listdir(in_dir):
+            logger.info(
+                "CMORizing var %s:%s for year %s", vals['mip'], var, year
+            )
+            var_info = cmor_table.get_variable(vals['mip'], var)
+            file_pattern = '{0}_{1}_{2}_*.nc'.format(
+                vals['raw'], hemisphere, vals['grid']
+            )
+            raw_info = {
+                'name': vals['raw'],
+                'file': os.path.join(in_dir, year, '??', file_pattern)
+            }
+            glob_attrs['mip'] = vals['mip']
+            extract_variable(
+                var_info, raw_info, out_dir, glob_attrs, year
+            )
+            if first_run:
+                sample_file = glob.glob(os.path.join(
+                    in_dir, year, '01', file_pattern))[0]
+                cube = iris.load_cube(
+                    sample_file,
+                    iris.Constraint(
+                        cube_func=lambda c: c.var_name == raw_info['name'])
+                )
+                _create_areacello(
+                    cfg,
+                    cube,
+                    glob_attrs,
+                    out_dir)
+                first_run = False
+
+
+def extract_variable(var_info, raw_info, out_dir, attrs, year):
+    """Extract to all vars."""
+    var = var_info.short_name
+    cubes = iris.load(
+        raw_info['file'],
+        iris.Constraint(cube_func=lambda c: c.var_name == raw_info['name'])
+    )
+
+    tracking_ids = _unify_attributes(cubes)
+    cube = cubes.concatenate_cube()
+    del cubes
+    if var_info.frequency == 'mon':
+        cube = _monthly_mean(cube)
+    if var_info.frequency == 'day':
+        if cube.coord('time').shape[0] < 300:
+            logger.warning(
+                'Only %s days available. Can not generate daily files',
+                cube.coord('time').shape[0]
+            )
+            return
+    logger.debug(cube)
+    if tracking_ids:
+        cube.attributes['tracking_ids'] = tracking_ids
+    cube.coord('projection_x_coordinate').var_name = 'x'
+    cube.coord('projection_y_coordinate').var_name = 'y'
+    lon_coord = cube.coord('longitude')
+    lon_coord.points = _correct_lons(lon_coord.points)
+
+    fix_var_metadata(cube, var_info)
+    convert_timeunits(cube, year)
+    set_global_atts(cube, attrs)
+    save_variable(cube, var, out_dir, attrs)
+    return cube
+
+
+@vectorize(['float32(float32)'])
```

Comment:
Can you use numpy instead? I would like to try to get rid of our numba dependency at some point, because we hardly use it. I think you can do something like `lon[lon < 0] += 360` to do this in numpy.